src/snowflake/connector/ocsp_snowflake.py

```diff
@@ -1322,13 +1373,23 @@ def _validate_certificates_sequential(
             )
 
         for issuer, subject in cert_data:
-            r = self.validate_by_direct_connection(
-                issuer, subject, telemetry_data, hostname, do_retry=do_retry
-            )
-            results.append(r)
+            cert_id, _ = self.create_ocsp_request(issuer=issuer, subject=subject)
+            cache_key = generate_cache_key(cert_id, subject)
+            cache_hit = OCSP_CACHE.get(cache_key)
+            if cache_hit is None:
```

Comment:
I'm curious if we plan to support a `loader` concept (similar to Guava's loading cache) so if a key is not present in the cache, the cache triggers loader and returns it, and also caches the loaded result for next call.

Reply:
This PR only introduces an in-memory cache, so not yet. I'm still flushing out the exact save/load strategies.
I'll keep this is mind for the final strategies though!